backend/routes/admin.py

```
@@ -0,0 +1,656 @@
+            httponly=is_production,  # Allow JS access in development
+            secure=is_production,  # Only secure in production
+            samesite="lax",  # Lax is better for same-domain dev
+        )
+
+        return LoginResponse(
+            success=True, message="Login successful", user=user_data, session_id=auth_result["session_id"]
+        )
+
+    except Exception as e:
+        logger.error(f"Login error: {str(e)}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Login failed")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)